swift/ql/lib/codeql/swift/elements/decl/MethodDecl.qll

```diff
@@ -21,4 +21,33 @@ class MethodDecl extends AbstractFunctionDecl {
     or
     this = getAMember(any(ProtocolDecl c))
   }
+
+  /**
+   * Holds if this function is called `funcName` and its a member of a
+   * class, struct, extension, enum or protocol call `typeName`.
+   */
+  predicate hasQualifiedName(string typeName, string funcName) {
+    this.getName() = funcName and
+    (
+      exists(NominalTypeDecl c |
+        c.getFullName() = typeName and
+        c.getAMember() = this
+      )
+      or
+      exists(ExtensionDecl e |
+        e.getExtendedTypeDecl().getFullName() = typeName and
+        e.getAMember() = this
+      )
+    )
+  }
+
+  /**
+   * Holds if this function is called `funcName` and its a member of a
+   * class, struct, extension, enum or protocol call `typeName` in a module
+   * called `moduleName`.
+   */
+  predicate hasQualifiedName(string moduleName, string typeName, string funcName) {
+    this.hasQualifiedName(typeName, funcName) and
+    this.getModule().getFullName() = moduleName
```

Comment:
I think the `getFullName` on `Module` should probably also be cached, but we can worry about that later 👍.

Reply:
Seems reasonable.  Done as well.